bugsnag-spring/src/common/java/com/bugsnag/ScheduledTaskConfiguration.java

```diff
@@ -87,33 +111,87 @@ private TaskScheduler createNewTaskScheduler(
     }
 
     /**
-     * If a task scheduler has been defined by the application, get it so that
-     * bugsnag error handling can be added.
-     * <p>
-     * Reflection is the simplest way to get and set an error handler
-     * because the error handler setter is only defined in the concrete classes,
-     * not the TaskScheduler interface.
-     *
-     * @param taskScheduler the task scheduler
+     * If a task scheduler has been defined by the application, configure it so that Bugsnag error handling is added.
+     * We first capture any existing ErrorHandler (e.g. your mock bean), chain it into the Bugsnag handler,
+     * then set the Bugsnag handler via setter if available (Boot 3+) or via field reflection.
      */
-    private void configureExistingTaskScheduler(TaskScheduler taskScheduler,
-                                                BugsnagScheduledTaskExceptionHandler errorHandler) {
+    private void configureExistingTaskScheduler(
+            TaskScheduler taskScheduler,
+            BugsnagScheduledTaskExceptionHandler errorHandler
+    ) {
+        // (1) Capture whatever handler is already configured on the scheduler
+        ErrorHandler existing = extractExistingErrorHandler(taskScheduler);
+        if (existing != null) {
+            errorHandler.setExistingErrorHandler(existing);
+        } else if (scheduledTaskErrorHandlerBean != null) {
+            // Fallback: chain the bean-level handler if the scheduler didn't have one yet
+            errorHandler.setExistingErrorHandler(scheduledTaskErrorHandlerBean);
+        }
+
+        // (2) Install the Bugsnag handler via public setter if available, else via private field
+        if (trySetErrorHandlerViaMethod(taskScheduler, errorHandler)) {
+            return;
+        }
+        trySetErrorHandlerViaField(taskScheduler, errorHandler);
```

Comment:
the `return` here is a bit unexpected I feel, the code might read more easily as:

```suggestion
        if (!trySetErrorHandlerViaMethod(taskScheduler, errorHandler)) {
            trySetErrorHandlerViaField(taskScheduler, errorHandler);
        }
```